src/api/utils/fetch-factory.ts

```diff
@@ -0,0 +1,127 @@
+import { FetchInterceptorManager } from './fetch-interceptor-manager'
+
+export interface FetchConfig {
+  baseUrl?: string
+  defaultHeaders?: HeadersInit
+}
+
+function buildUrl (url: string, baseUrl: string): string {
+  if (url.startsWith('http://') || url.startsWith('https://')) {
+    return url
+  }
+
+  const cleanBaseUrl = baseUrl.replace(/\/$/, '')
+  const cleanPath = url.startsWith('/') ? url : `/${url}`
+
+  return `${cleanBaseUrl}${cleanPath}`
+}
+
+async function getBody<T> (response: Response): Promise<T> {
+  const contentType = response.headers.get('content-type')
+
+  const contentTypeHandlers: Record<string, (response: Response) => Promise<any>> = {
+    'application/json': response => response.json(),
+    'application/pdf': response => response.blob(),
+    'application/octet-stream': response => response.arrayBuffer(),
+    'text': response => response.text(),
+  }
+
+  const defaultHandler = (response: Response) => response.text()
+
+  const getHandler = (contentType: string | null) => {
+    if (!contentType) {
+      return defaultHandler
+    }
+
+    for (const [type, handler] of Object.entries(contentTypeHandlers)) {
+      if (contentType.includes(type)) {
+        return handler
+      }
+    }
+
+    return defaultHandler
+  }
```

Comment:
Toute cette logique peut être exportée dans une méthode propre. Tout rassembler dans `getHandler` et le sortir de `getBody`

```suggestion
  const getHandler = (contentType: string | null) => {
    const contentTypeHandlers: Record<string, (response: Response) => Promise<any>> = {
      'application/json': response => response.json(),
      'application/pdf': response => response.blob(),
      'application/octet-stream': response => response.arrayBuffer(),
      'text': response => response.text(),
    }
    const defaultHandler = (response: Response) => response.text()
  
    if (!contentType) {
      return defaultHandler
    }

    for (const [type, handler] of Object.entries(contentTypeHandlers)) {
      if (contentType.includes(type)) {
        return handler
      }
    }

    return defaultHandler
  }
```